common/models/acl.js

```diff
@@ -204,11 +205,12 @@ module.exports = function(ACL) {
   /*!
    * Resolve permission from the ACLs
    * @param {Object[]) acls The list of ACLs
-   * @param {Object} req The request
-   * @returns {AccessRequest} result The effective ACL
+   * @param {AccessRequest} req The access request
+   * @returns {AccessRequest} result The resolved access request
    */
   ACL.resolvePermission = function resolvePermission(acls, req) {
     if (!(req instanceof AccessRequest)) {
+      req.registry = this.registry;
       req = new AccessRequest(req);
```

Comment:
When specifying an "arg": "options" in a common/models/xxx.json 'accepts' section, we ran into problems, finding 'authorizedRoles: {}' added to what was intended to be a list of options to pass to a system call, causing problems since we are simply passing the options values forward (in which case authorizedRoles becomes an invalid option). Our workaround was to rename to arg from "options" to something else (eg "lsfOptions".  It would be more convenient to not have to worry about this; without knowing the inner workings, having the authorizedRoles in some more uniquely named structure would help the situation.